### ISSUE TITLE
Idea: add component grouping in main view

### ISSUE THREAD

**Issue by cr1tbit** [Bouni/kicad-jlcpcb-tools#504]
![image](https://github.com/user-attachments/assets/5cb29b93-9236-4fc9-8960-3f4c15042c91)

I find myself always clicking "select alike parts" during JLC component selection. I understand that there are cases where you want to manually replace a single component here, but the defaut mode is choosing the same LCSC number for all of them.

There should be another "option" on the right bar, that will allow grouping all matching components.

Another argument - due to WX limitation, after coming back from component selection submenu, the list is always fully redrawn - because of that, the view will always go to the top. With large BOM, this becomes a bit annoying quite fast. Grouping components would not fix this directly, but the list will become a lot shorter, so the scrollback issue will not be that bad then.

**Comment by chmorgan** [comment]
Yeah the idea of 'auto select alike parts' would be a cool feature to add, and/or grouping by alike parts, otherwise its hard to see how many 'alike' parts that were selected.

**Comment by Bouni** [comment]
You're are totally right, I want to refactor the component list for a long time but have no idea how to achieve this.

PRs are welcome 😉

**Comment by z2amiller** [comment]
The 'Select Alike Parts' button is now a toggle, so as long as it is activated, you'll label all selected parts by default.


Fixed by: #712
